Enhancing push

mmkekic opened the pull request:
This PR addresses issues #4 (adding a test) and #5.

mmkekic committed:
Add test for implicit pipe inside push method.

mmkekic committed:
Modify push to accept tuple as pipe argument.

mmkekic committed:
Make source and pipe keyword arguments in push.

gonzaponte commented:
I think we also want a test to ensure the arguments are keyword-only.

jacg commented:
> I think we also want a test to ensure the arguments are keyword-only.

Yes, please.

I've rebased this to resolve conflicts with #3, and pushed to my repo, so you might like to take it from there.

mmkekic committed:
Add test for push arguments being keyword only.

The tests checks if TypeError is raised in case push arguments are
given as positional.
